### PR TITLE
[TT-4757] Refactor jwt OAS conversion

### PR DESCRIPTION
**Pull request by furkansenharputlu**

### CODE REVIEW

**Review by jeffy-mathew — APPROVED** [review]

File: apidef/oas/oas.go [hunk 1]
```diff
@@ -111,6 +111,28 @@ func (s *OAS) getTykTokenAuth(name string) (token *Token) {
...
 }
 
+func (s *OAS) getTykJWTAuth(name string) (jwt *JWT) {
+	if securitySchemes := s.getTykSecuritySchemes(); securitySchemes != nil {
+		securityScheme := securitySchemes[name]
+		if securityScheme == nil {
+			return
+		}
+
+		mapSecurityScheme, ok := securityScheme.(map[string]interface{})
```
[the remark this answers is not in the document]
Comment (titpetric) [reply apidef/oas/oas.go:121]: Or to ask it in another way: is securityScheme ever anything else than map[string]interface{}?

File: apidef/oas/security_test.go [hunk 2]
```diff
@@ -284,3 +284,48 @@ func TestOAS_AppendSecurity(t *testing.T) {
...
+
+	var api apidef.APIDefinition
+	api.AuthConfigs = make(map[string]apidef.AuthConfig)
+	oas.extractJWTTo(&api, securityName)
+
+	var convertedOAS OAS
+	convertedOAS.SetTykExtension(&XTykAPIGateway{Server: Server{Authentication: &Authentication{SecuritySchemes: map[string]interface{}{}}}})
+	convertedOAS.fillJWT(api)
+
+	assert.Equal(t, oas, convertedOAS)
```
Comment (titpetric) [inline apidef/oas/security_test.go:330]: Can we add a test for failure, say, an invalid parameter for the security schemes? Disregard if it already exists.

**Review by titpetric — APPROVED** [review]

File: apidef/oas/oas.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (furkansenharputlu) [reply apidef/oas/oas.go:121]: Yes, it can be an object Token, JWT etc.

File: apidef/oas/oas.go [hunk 3]
```diff
@@ -111,6 +111,28 @@ func (s *OAS) getTykTokenAuth(name string) (token *Token) {
...
+		if ok {
+			jwt = &JWT{}
+			inBytes, _ := json.Marshal(mapSecurityScheme)
+			_ = json.Unmarshal(inBytes, jwt)
```
[the remark this answers is not in the document]
Comment (furkansenharputlu) [reply apidef/oas/oas.go:125]: As we have done, json validation checks in the api handlers, these are not really possible to fail.

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/3934/merge
Commit: 3fd9937c1c0c1490a97d4ae37e91631cc18fc97e 
Triggered by: pull_request (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/1968822529)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/3934/merge
Commit: d382897bcb622bcace76a516095bc6d894dc9c7a 
Triggered by: pull_request (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/1968919795)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/heads/master
Commit: b405a7d2a61b1e7695f57a5c9cfa9ab515932d2a Refactor jwt OAS conversion (#3934)
Triggered by: push (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/1969079785)

**Pull request merged**